src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/osgi/OsgiFramework.kt

```diff
@@ -145,6 +203,28 @@ class OsgiFramework(sync: InstanceSync) : InstanceService(sync) {
         }
     }
 
+    private fun getConfigList(): List<ConfigurationState> {
+        var configList: List<ConfigurationState> = listOf()
+        val regex = Regex("configData = (.*);")
+        sync.http.get(CONFIGURATION_PATH) { response ->
+            val mapper = ObjectMapper()
+            val html = asString(response)
+            val configJson = regex.find(html)?.groups?.get(1)?.value ?: "{}"
+            val json = mapper.readTree(configJson)
+            configList = mapper.readValue(json.get("pids").toString())
+        }
+        return configList
+    }
+
+    private fun getConfigPayload(configuration: OSGiConfiguration, properties: Map<String, Any>): Map<String, Any> {
+        return properties.toMutableMap().also { payload ->
```

Comment:
why are you mutating / converting map? what is being done here?

Reply:
looks strange, document method behavior

Reply:
Changed method signature to pass `MutableMap`

Reply:
mutable things should not be args ;) general rule for immutablity

Reply:
Fixed